atd-vze/src/views/Crashes/Maps/CrashEditCoordsMap.js

```diff
@@ -1,189 +1,129 @@
-import React, { Component } from "react";
-import { withApollo } from "react-apollo";
-import { UPDATE_COORDS } from "../../../queries/crashes";
+import React from "react";
 
 import MapGL, {
   Marker,
   NavigationControl,
   FullscreenControl,
 } from "react-map-gl";
-import "mapbox-gl/dist/mapbox-gl.css";
-// import Geocoder from "react-map-gl-geocoder";
-// import { CustomGeocoderMapController } from "./customGeocoderMapController";
-import "react-map-gl-geocoder/dist/mapbox-gl-geocoder.css";
 
 import Pin from "./Pin";
+import { useMutation } from "react-apollo";
+import { useAuth0 } from "../../../auth/authContext";
+import { UPDATE_COORDS } from "../../../queries/crashes";
 import { CrashEditLatLonForm } from "./CrashEditLatLonForm";
-
 import {
-  LOCATION_MAP_CONFIG,
+  defaultInitialState,
   LabeledAerialSourceAndLayer,
+  isDev,
+  mapParameters,
 } from "../../../helpers/map";
 
-const TOKEN = process.env.REACT_APP_MAPBOX_TOKEN;
-
-const fullscreenControlStyle = {
-  position: "absolute",
-  top: 0,
-  left: 0,
-  padding: "10px",
-};
-
-const navStyle = {
-  position: "absolute",
-  top: 36,
-  left: 0,
-  padding: "10px",
-};
-
-// const customGeocoderMapController = new CustomGeocoderMapController();
-
-class CrashEditCoordsMap extends Component {
-  constructor(props) {
-    super(props);
-
-    // Default map center
-    this.initialMapCenter = {
-      latitude: this.props.data.latitude_primary || 30.26714,
-      longitude: this.props.data.longitude_primary || -97.743192,
-    };
-
-    this.state = {
-      viewport: {
-        latitude: this.initialMapCenter.latitude,
-        longitude: this.initialMapCenter.longitude,
-        zoom: 17,
-        bearing: 0,
-        pitch: 0,
-      },
-      popupInfo: null,
-      markerLatitude: 0,
-      markerLongitude: 0,
-      isDragging: false,
-    };
-  }
-
-  // Tie map and geocoder control together
-  mapRef = React.createRef();
-
-  _handleViewportChange = viewport => {
-    this.setState({
-      viewport: { ...this.state.viewport, ...viewport },
-    });
-  };
-
-  _updateViewport = viewport => {
-    this.setState({
-      viewport,
-      markerLatitude: viewport.latitude,
-      markerLongitude: viewport.longitude,
-    });
+const CrashEditCoordsMap = ({
+  data,
+  mapGeocoderAddress,
+  crashId,
+  refetchCrashData,
+  setIsEditingCoords,
+}) => {
+  const { user } = useAuth0();
+  const { email = null } = user;
```

Comment:
oo i dont think ive seen this syntax before, could you explain whats going on here?

Reply:
definitely! destructuring allows for default values to be assigned in case that a value isn't present. Pretty handy but I've stumbled when working with deeply nested objects before https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#default_value